sync/syncer.go

```diff
@@ -273,19 +285,27 @@ func (s *Syncer) synchronise() {
 
 	//release synchronise lock
 	defer s.syncLock.Unlock()
+	curr := s.GetCurrentLayer()
+	weaklySyncedStatus := s.weaklySynced(curr)
 
 	//node is synced and blocks from current layer hav already been validated
-	if s.GetCurrentLayer() == s.lValidator.ProcessedLayer() {
+	if curr == s.ProcessedLayer() {
 		s.Debug("node is synced ")
 
 		// fully-synced, make sure we listen to p2p
 		s.setGossipBufferingStatus(Done)
 		return
 	}
 
+	if !weaklySyncedStatus && s.getGossipBufferingStatus() == Done {
+		s.SetZeroBlockLayer(s.GetCurrentLayer() - 1)
```

Comment:
Are we always guaranteed to get into this logic with only one unprocessed layer behind? I thick that it can be very subtle. If someone's computer gets into sleep and he wakes up after a while, they may have gossip open but no blocks, no?

Reply:
Considering both comments,  wouldn't it be better to run `s.SetZeroBlockLayer(s.ProcessedLayer() + 1)` as you do below?

Reply:
i want to wait sufficient time for us to decide that no blocks have arrived, maybe i should do it after propagation delta (30 sec's) like the validation

Reply:
done